tuf_conformance/metadata.py

```diff
@@ -28,7 +29,7 @@
 
 class MetadataTest(Metadata):
     @classmethod
-    def from_dict(cls, metadata: Dict[str, Any]) -> "MetadataTest[T]":
+    def from_dict(cls, metadata: Dict[str, Any]) -> "Metadata[T]":
```

Comment:
What is the reason for this change?

Reply:
mypy complains about this type annotation when the method is called.. and I believe it is correct to complain: we do return a Metadata from the is method because we end up calling `Metadata()` at the end of the function.

Reply:
I could be wrong about the logic there, python inheritance has never been my strongest suit

Reply:
the error is 
> tuf_conformance/metadata.py:32: error: "MetadataTest" expects no type arguments, but 1 given  [type-arg]

which admittedly is a bit strange.

Reply:
hmmm... I think this might be related to the generic type, let me try a different tweak to fix this

Reply:
Yeah tweaking the inheritance annotation `class MetadataTest(Metadata[T])` makes mypy understand the intent as well. Fixed with a new commit.